js/source/worker.js

```diff
@@ -21,13 +15,31 @@ module.exports = function(self) {
 function Worker(self) {
     this.self = self;
     this.actor = new Actor(self, this);
-    this.loading = {};
 
+    this.loading = {};
     this.loaded = {};
-    this.geoJSONIndexes = {};
+
+    this.plugins = {};
+    self.registerPlugin = function (name, plugin) {
```

Comment:
Is "plugin" the right word here? To me, plugin implies some really generic module, whereas these are specifically source types. Would `registerSourceType` be a good name?